### PR TITLE
koordlet: add close for querier

### PR DESCRIPTION
**Pull request by zwzhang0107**
### Ⅱ. Does this pull request fix one issue?

#1981

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

**Review by jasonliu747 — APPROVED** [review]

File: pkg/koordlet/qosmanager/helpers/metrics_query.go [hunk 1]
```diff
@@ -63,6 +63,8 @@ func CollectAllHostAppMetrics(hostApps []slov1alpha1.HostApplicationSpec, metric
 	queryParam metriccache.QueryParam, metricResource metriccache.MetricResource) map[string]float64 {
 	appsMetrics := make(map[string]float64)
 	querier, err := metricCache.Querier(*queryParam.Start, *queryParam.End)
+	defer querier.Close()
```
Comment (saintube) [inline pkg/koordlet/qosmanager/helpers/metrics_query.go:66]: may repeatedly close the querier while invoking querier.Query()

File: pkg/koordlet/metriccache/tsdb_storage.go [hunk 2]
```diff
@@ -199,3 +207,14 @@ func (t *tsdbQuerier) Query(meta MetricMeta, hints *QueryHints, result MetricRes
...
 }
+
+func (t *tsdbQuerier) QueryAndClose(meta MetricMeta, hints *QueryHints, result MetricResult) error {
+	defer t.Close()
+	return t.Query(meta, hints, result)
+}
+
+func (t *tsdbQuerier) Close() {
+	if err := t.querier.Close(); err != nil {
+		klog.Warningf("close host application querier failed, error: %v", err)
```
Comment (saintube) [inline pkg/koordlet/metriccache/tsdb_storage.go:218]: wrong message "host application"

**Review by saintube — APPROVED** [review]
/lgtm

**Pull request merged**